yodeploy/cmds/install_certificates.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import stat
+import sys
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))
+
+from yodeploy import config  # noqa
+
+list_path = 'envs/list/'
+cert_path = 'envs/certificate/public'
+key_path = 'envs/certificate/private'
+
+
+def find_ssl_path(settings):
+    ssl_dir = os.path.expanduser(settings.yodeploy.ssl_path)
+    certs_dir = os.path.join(ssl_dir, 'certs')
+    private_dir = os.path.join(ssl_dir, 'private')
+    if not os.path.exists(ssl_dir):
+        os.makedirs(ssl_dir)
+    if not os.path.exists(certs_dir):
+        os.mkdir(certs_dir)
+    if not os.path.exists(private_dir):
+        os.mkdir(private_dir)
+        os.chmod(private_dir, stat.S_IRWXU)
+    return ssl_dir
+
+
+def remove_all_files(path):
+    for f in os.listdir(path):
+        if os.path.isfile(f):
+            print('removing {}'.format(f))
+            os.unlink(f)
+
+
+def clean_ssl_directory(settings):
+    ssl_dir = os.path.expanduser(settings.yodeploy.ssl_path)
+    certs_dir = os.path.join(ssl_dir, 'certs')
+    private_dir = os.path.join(ssl_dir, 'private')
+    remove_all_files(certs_dir)
+    remove_all_files(private_dir)
+
+
+def get_certificate(url, name, username, password):
+    cert_url = url + '/'.join([cert_path, name])
```

Comment:
Taking a step back for just a second here, this implementation could be cleaned up significantly by dropping in demands and using it to define a very simple client. All of the request building would be minimized and you could rip out the error handling in the command body because demands will blow up by default if the requests fail: https://github.com/yola/demands

```python
from demands import HTTPServiceClient

class CertificateService(HTTPServiceClient):
    def get_certificate(self, env_name, default_filename):
        response = self.get('/envs/certificate/public/{env_name}/'.format(env_name))
        return get_filename_from_headers(response, default_filename), response.content

    def get_key(self, env_name):
        pass

    def get_environments(self):
        pass

service = CertificateService(
    url='http://certificateservice.tld/',
    auth=('username','pass'),
)

certificate = service.get_certificate('snowman')  # raises exception if the request fails
```

That said, I'd have to defer to @stefanor on whether or not we want to add another dependency here.